app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+  
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+  validates_presence_of :password, :username, :name, :email
```

Comment:
Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.